### PR TITLE
Build/chain management

### PR DESCRIPTION
**Pull request by ccamel**
This Pull Request adds a set of tasks to the [Makefile](https://sagiegurari.github.io/cargo-make/) that makes it easy and convenient for developers to manage the blockchain and deploy Smart Contracts.

These tasks allow for quick and reproducible initialization of the blockchain with default settings, including the creation of 4 pre-funded wallets (1 for the validator and 3 for testing). The tasks also enable the management of the blockchain's lifecycle, including starting, stopping, and log inspection, as well as the deployment of CosmWasm contracts.

The implementation relies on Docker and the latest available official [wasmd](https://hub.docker.com/r/cosmwasm/wasmd) image for the blockchain.

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -117,6 +117,225 @@ echo "📖 Documentation has been successfully generated and available at $(pwd)
...
+      --home ${CHAIN_HOME} \
+      --chain-id ${CHAIN}
+'''
+
+[tasks.chain-inspect-contract]
+description = "Inspect a specific contract deployed to the chain."
+script = '''
+if [ -z "$1" ]
+then
+  echo "❌ Please provide the constract code ID as the first argument."
```
Comment (bdeneux) [inline Makefile.toml:325]:
```suggestion
  echo "❌ Please provide the contract code ID as the first argument."
```

**Review by bdeneux — APPROVED** [review]
Wonderful 😍. 

Why we doesn't use the okp4d docker image ? If we add contacts that use modules included in the okp4 images we will not be able to test it with those tools ?

**Comment by ccamel** [comment]
> Why we doesn't use the okp4d docker image ? If we add contacts that use modules included in the okp4 images we will not be able to test it with those tools ?

Yes, I have asked myself that question. And you're right, some smart contracts will have to use the `logic` module of the okp4 chain - currently under development. So...

What I'd suggest is to accept the PR as is, with the official `wasmd` image, and in another PR make the proposal of a move to `okp4d`. If you agree, I propose that you take charge of this?

**Review by hleonardtek — APPROVED** [review]

**Review by bdeneux — APPROVED** [review]

**Comment by bdeneux** [comment]
> > Why we doesn't use the okp4d docker image ? If we add contacts that use modules included in the okp4 images we will not be able to test it with those tools ?
> 
> Yes, I have asked myself that question. And you're right, some smart contracts will have to use the `logic` module of the okp4 chain - currently under development. So...
> 
> What I'd suggest is to accept the PR as is, with the official `wasmd` image, and in another PR make the proposal of a move to `okp4d`. If you agree, I propose that you take charge of this?

Yes I understand, merge this PR and I will do that in another PR :).

**Review by ad2ien — APPROVED** [review]
Tested on linux: scripts don't have permission to write into subfolders created by docker so I added a folder initialization step. Otherwise everything works like a charm 😀

**Review by bdeneux — APPROVED** [review]

**Review by hleonardtek — APPROVED** [review]

**Comment by ccamel** [comment]
Let's go. Thanks to all!

**Pull request merged**